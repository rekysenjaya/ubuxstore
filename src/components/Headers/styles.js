
import { StyleSheet, Dimensions } from 'react-native';

const D = Dimensions.get('window')

const styles = StyleSheet.create({
    header: {
        backgroundColor: '#F17C5F',
        height: 70,
        justifyContent: 'center'
    },
    headerContent: {
        flexDirection: 'column',
        width: D.width - 20
    },
    headerContentBody: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'center'
    },
    headerContentIcon: {
        marginHorizontal: 10,
        justifyContent: 'center'
    },
    title: {
        color: '#fff',
        fontWeight: '900',
        fontSize: 20
    },
    badge: {
        position: 'absolute',
        zIndex: 1,
        top: 13,
        right: -8
    },
    input: {
        marginHorizontal: 10,
        marginBottom: 10,
        flex: 1,
        justifyContent: 'center'
    },
    iconCart: {
        color: 'white',
        marginTop: 20
    },
})

export default styles