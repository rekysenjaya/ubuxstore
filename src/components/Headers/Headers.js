import React, { Component } from 'react'
import { Alert } from 'react-native'
import { Container, Content, Header, Icon, View, Badge, Text } from 'native-base'
import { TouchableOpacity } from 'react-native'
import styles from './styles'
import PropTypes from 'prop-types'

export default class Headers extends Component {
    render() {
        return (
            <Container>
                <Header style={styles.header} androidStatusBarColor="#F17C5F" >
                    <View style={styles.headerContent} >
                        <View style={styles.headerContentBody} >
                            <TouchableOpacity style={styles.headerContentIcon} onPress={() => this.props.goBack()} >
                                <Icon name="arrow-back" style={{ color: 'white' }} />
                            </TouchableOpacity>
                            <View rounded style={styles.input} >
                                <Text style={styles.title} >{this.props.title || 'Store Details'}</Text>
                            </View>
                            {(this.props.cart && Boolean(this.props.cart.length)) && (<TouchableOpacity
                                style={styles.badge}
                                onPress={() => this.props.navigate()} >
                                <Badge>
                                    <Text>{this.props.cart.length}</Text>
                                </Badge>
                            </TouchableOpacity>)}
                            {(this.props.clearCart) && (<TouchableOpacity
                                style={styles.headerContentIcon}
                                onPress={() => Alert.alert(
                                    'Delete All Item',
                                    `you are sure to delete all?`,
                                    [
                                        { text: 'No', style: 'cancel' },
                                        { text: 'Yes', onPress: () => this.props.clearCart() },
                                    ],
                                    { cancelable: false }
                                )} >
                                <Badge >
                                    <Text>Delete ALL</Text>
                                </Badge>
                            </TouchableOpacity>)}
                            {this.props.cart && <TouchableOpacity style={styles.headerContentIcon} onPress={() => this.props.navigate()} >
                                <Icon name="cart" style={styles.iconCart} />
                            </TouchableOpacity>}
                        </View>
                    </View>
                </Header>
                <Content>
                    {this.props.children}
                </Content>
            </Container>
        )
    }
}

Headers.propTypes = {
    title: PropTypes.string,
    cart: PropTypes.array,
    goBack: PropTypes.func,
    clearCart: PropTypes.func,
    navigate: PropTypes.func,
    children: PropTypes.array,
}
