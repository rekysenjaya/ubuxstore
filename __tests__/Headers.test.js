/**
 * Sample React Native Snapshot Test
 */

'use strict';

import 'react-native';
import React from 'react';
import Headers from '../src/components/Headers';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.setTimeout(15000);

it('renders correctly', () => {
    const tree = renderer.create(<Headers
        cart={[]}
        children={[]}
        title=""
        goBack={() => console.log('')}
        navigate={() => console.log('')} />).toJSON();
    expect(tree).toMatchSnapshot();
});
