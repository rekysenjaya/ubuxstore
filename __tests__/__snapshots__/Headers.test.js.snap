// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      androidStatusBarColor="#F17C5F"
      style={
        Array [
          Object {
            "backgroundColor": "#F8F8F8",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 6,
            "paddingRight": 10,
            "paddingTop": 18,
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "top": 0,
          },
          Object {
            "backgroundColor": "#F17C5F",
            "height": 70,
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {
              "flexDirection": "column",
              "width": 730,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "justifyContent": "center",
                "marginHorizontal": 10,
                "opacity": 1,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "white",
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            rounded={true}
            style={
              Array [
                Object {},
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                  "marginBottom": 10,
                  "marginHorizontal": 10,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#fff",
                    "fontSize": 20,
                    "fontWeight": "900",
                  },
                ]
              }
              uppercase={false}
            >
              Store Details
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "justifyContent": "center",
                "marginHorizontal": 10,
                "opacity": 1,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "white",
                      "marginTop": 20,
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    viewIsInsideTabBar={false}
  >
    <View />
  </RCTScrollView>
</View>
`;
